### PR TITLE
Fix sequencing of MakeCommonResourcesPriFile

### PR DESCRIPTION
**Pull request by ViktorHofer**
Requires https://github.com/dotnet/arcade/pull/3102

This also makes the allconfigurations leg faster as it disables binplacing and pri generation on it.

### CODE REVIEW

File: src/tests.proj [hunk 1]
```diff
@@ -8,6 +8,7 @@
     <CoverageReportDir>$(ArtifactsDir)coverage</CoverageReportDir>
     <!-- For UAP we are using an APPX that is registered with a unique ID. Because of that we need to run tests sequentially -->
     <SerializeProjects Condition="'$(BuildingUAPVertical)' == 'true' or '$(Coverage)' == 'true'">true</SerializeProjects>
+    <TraversalBuildDependsOn Condition="'$(BuildingUAPVertical)' == 'true' and '$(BuildAllConfigurations)' != 'true'">MakeCommonResourcesPriFile;$(TraversalBuildDependsOn)</TraversalBuildDependsOn>
```
Comment (ericstj) [inline src/tests.proj]: Did you consider putting this in https://github.com/dotnet/corefx/blob/master/src/pretest.builds?

Comment (ViktorHofer) [reply src/tests.proj]: That's probably a better place for it but I will keep it as is to unblock CI. We can move it later.

Comment (ericstj) [reply src/tests.proj]: Was thinking the same

**Comment by ViktorHofer** [comment]
Triggered a retry just to make sure that the restore issue on musl_x64 was a one-off thing.

**Comment by ViktorHofer** [comment]
Yes was transient. Merging.

**Pull request merged**